### DISCUSSION TITLE
How to control temperature and time step using yml files?

### DISCUSSION THREAD

**Discussion by eacoba** [Becksteinlab/MDPOW#211] [Q&A] [answered]
For thier simplicity, I prefer to use yml files to run mdpow. But I have some doubts about how to control temperature and time steps using yml files.

1.  I wanted to simulate at 310 K, so I edited the mdp files changing temperature, and simulation (relaxed and NPT) mdps are in 310 K. But, I'm not sure if this is also taking in account to calculate fep. I edited two scripts, in the following lines:

mdpow/fep.py
`486 self.Temperature = kwargs.pop('temperature', **300.0**)` ---> `486 self.Temperature = kwargs.pop('temperature', 310.0)`
mdpow/analysis.py
`179 def gsolv2logpow(Gwat, Goct, unit='kcal/mol', temperature=**300**.)` ---> `179 def gsolv2logpow(Gwat, Goct, unit='kcal/mol', temperature=310.)`
`638 temperature = kwargs.pop('temperature', **300.0**)` ---> `temperature = kwargs.pop('temperature', 310.0)` 

2.  Working with octanol NPT simulations, I had several problems with LINCS, so I decided to change time step to 1 fs using mdp files, but this does not change in simulation files. How is possible to do it?

I'm not sure if is important, but I'm using a pip installation.

**Comment by orbeckst** [comment, marked as answer]
The current code is not set up to change temperature or timestep via yml file. As you say, you have to use custom MDP files for each step. You are also correct that this leads to problems with the analysis because the `Temperature` in the .fep pickle file that is associated with the run is _not the same as in the MDP file_.

Thus, if you use mdpow with the scripts and yaml files and you changed the temperature then your analysis results will be incorrect.

I opened issue #266 .

----

Your solution is a temporary hack and will work for you specifically but it would be better to add entries to the runinput.yml file for temperature and propagate them correctly.

**Comment by orbeckst** [comment]
I opened issue #266. Thank you for the feedback. Sorry I hadn't seen it earlier.

**Comment by orbeckst** [comment]
I didn't realize that closing a discussion would hide it so I am reopening so that people can actually see it.
